netplugin/netd.go

```diff
@@ -242,6 +242,33 @@ func processEpState(netPlugin *plugin.NetPlugin, opts cliOpts, epID string) erro
 	return err
 }
 
+//processBgpEvent processes Bgp neighbor add/delete events
+func processBgpEvent(netPlugin *plugin.NetPlugin, opts cliOpts, hostID string,
+	isDelete bool) (err error) {
+
+	if opts.hostLabel != hostID {
+		log.Errorf("Ignoring Bgp Event on this host")
+		return
+	}
+	netPlugin.Lock()
+	defer func() { netPlugin.Unlock() }()
+
+	operStr := ""
+	if isDelete {
+		err = netPlugin.DeleteBgp(hostID)
+		operStr = "delete"
+	} else {
+		err = netPlugin.AddBgp(hostID)
+		operStr = "create"
+	}
+	if err != nil {
+		log.Errorf("Bgp operation %s failed. Error: %s", operStr, err)
+	} else {
+		log.Infof("Bgp operation %s succeeded", operStr)
+	}
+	return
```

Comment:
Btw, this i didn't add this code. I just moved this code from bottom of the file to middle where all the other event handlers are there. I'll be happy to make the return values more explicit.